opensrp-chw/src/main/java/org/smartregister/chw/interactor/FpUpcomingServicesInteractor.java

```diff
@@ -2,21 +2,113 @@
 
 import android.content.Context;
 
+import org.jeasy.rules.api.Rules;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.smartregister.chw.R;
 import org.smartregister.chw.anc.domain.MemberObject;
+import org.smartregister.chw.anc.domain.Visit;
 import org.smartregister.chw.anc.interactor.BaseAncUpcomingServicesInteractor;
 import org.smartregister.chw.anc.model.BaseUpcomingService;
+import org.smartregister.chw.core.rule.FpAlertRule;
+import org.smartregister.chw.core.utils.CoreConstants;
+import org.smartregister.chw.core.utils.FpUtil;
+import org.smartregister.chw.core.utils.HomeVisitUtil;
+import org.smartregister.chw.core.utils.VaccineScheduleUtil;
+import org.smartregister.chw.fp.dao.FpDao;
+import org.smartregister.chw.fp.domain.FpAlertObject;
+import org.smartregister.chw.fp.util.FamilyPlanningConstants;
 
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
 
 public class FpUpcomingServicesInteractor extends BaseAncUpcomingServicesInteractor {
-    private Flavor flavor = new FpUpcomingServicesInteractorFlv();
+    protected MemberObject memberObject;
+    protected Context context;
 
     @Override
-    protected List<BaseUpcomingService> getMemberServices(Context context, MemberObject memberObject) {
-        return flavor.getMemberServices(context, memberObject);
+    public List<BaseUpcomingService> getMemberServices(Context context, MemberObject memberObject) {
+        this.memberObject = memberObject;
+        this.context = context;
+        VaccineScheduleUtil.updateOfflineAlerts(memberObject.getBaseEntityId(), new DateTime(memberObject.getDob()), CoreConstants.SERVICE_GROUPS.CHILD);
+        List<BaseUpcomingService> serviceList = new ArrayList<>();
+        evaluateFp(serviceList);
+        return serviceList;
     }
 
-    public interface Flavor {
-        List<BaseUpcomingService> getMemberServices(Context context, MemberObject memberObject);
+    private void evaluateFp(List<BaseUpcomingService> serviceList) {
```

Comment:
- Should this method probably return the list of upcoming services? i.e. `List<BaseUpcomingService> serviceList` instead of return void yet it modifies the List. 
- If so, then maybe renaming it to getUpcomingServicesList or something similar would make what it does more explicit
- Then lastly, there won't be a need to pass in an empty list. The new list would be created inside the method instead.

Reply:
The method evaluates if there are any upcoming services and if they are there it then updates the List. That is why it's void. Main purpose is to just evaluate